pkg/asset/installconfig/powervs/session.go

```diff
@@ -185,19 +187,31 @@ func (c *BxClient) ValidateAccountPermissions() error {
 }
 
 //ValidateDhcpService checks for existing Dhcp service for the provided PowerVS cloud instance
-func (c *BxClient) ValidateDhcpService(ctx context.Context, svcInsID string) error {
+func (c *BxClient) ValidateDhcpService(ctx context.Context, svcInsID string, machineNetworks []types.MachineNetworkEntry) error {
 	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
 	defer cancel()
 
-	//create Power VS DHCP Client
-	dhcpClient := instance.NewIBMPIDhcpClient(ctx, c.PISession, svcInsID)
-	//Get all DHCP Services
-	dhcpServices, err := dhcpClient.GetAll()
+	// Create Power VS Network Client
+	networkClient := instance.NewIBMPINetworkClient(ctx, c.PISession, svcInsID)
+	// Get all network references
+	networkReferences, err := networkClient.GetAll()
 	if err != nil {
-		return errors.Wrap(err, "failed to get DHCP service details")
-	}
-	if len(dhcpServices) > 0 {
-		return fmt.Errorf("DHCP service already exists for provided cloud instance")
+		return errors.Wrap(err, "failed to get network references")
+	}
+	// Check each machineNetwork, typically one
+	for _, machineNetwork := range machineNetworks {
+                _, n1, _ := net.ParseCIDR(machineNetwork.CIDR.String())
+		// Use our networkReferences to get networks
+		for _, networkRef := range networkReferences.Networks {
+			network, err := networkClient.Get(*networkRef.NetworkID)
```

Comment:
Is this `Get` necessary? You already got all of the network references above. You should already have that information you are looping over.

Reply:
I believe so, as the get all from the NetworkClient:
https://github.com/openshift/installer/blob/99b9d7b23a2308cddf3965de306fadfe60c6097b/vendor/github.com/IBM-Cloud/power-go-client/clients/instance/ibm-pi-network.go#L43

Returns a Networks (plural) Object
https://github.com/openshift/installer/blob/99b9d7b23a2308cddf3965de306fadfe60c6097b/vendor/github.com/IBM-Cloud/power-go-client/power/models/networks.go#L21

Which is just an array of NetworkReference objects, 
https://github.com/openshift/installer/blob/99b9d7b23a2308cddf3965de306fadfe60c6097b/vendor/github.com/IBM-Cloud/power-go-client/power/models/network_reference.go#L21

We need a cidr, which seems to come from a Network (singular) object
https://github.com/openshift/installer/blob/99b9d7b23a2308cddf3965de306fadfe60c6097b/vendor/github.com/IBM-Cloud/power-go-client/power/models/network.go#L22

To get that Network (Singular) object we can call a Get against our NetworkClient, which requires a Network id
https://github.com/openshift/installer/blob/99b9d7b23a2308cddf3965de306fadfe60c6097b/vendor/github.com/IBM-Cloud/power-go-client/clients/instance/ibm-pi-network.go#L28

I could be a complete idiot and missing something obvious, but I don't think we have our Network (singular) objects from our GetAll call against our NetworkClient.

Reply:
@barbacbd Based on the above do you agree that a Get seems to be necessary here?

Reply:
@barbacbd I looked into this and I agree that the Get is necessary. The NetworkReference objects do not provide subnet CIDRs, so you have to get based on the ID to determine it.